File: Documentation/5.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Documents/Put.cs

using Raven.Client.Documents;
using Raven.Client.Documents.Commands;
using Raven.Client.Documents.Session;
using Raven.Documentation.Samples.Orders;
using Sparrow.Json;

namespace Raven.Documentation.Samples.ClientApi.Commands.Documents
{
    public class PutInterfaces
    {
        private class PutDocumentCommand
        {
            #region put_interface
            public PutDocumentCommand(string id, string changeVector, BlittableJsonReaderObject document)
            #endregion
            {
            }
        }
    }

    public class PutSamples
    {
        public void Foo()
        {
            using (var store = new DocumentStore())
            using (var session = store.OpenSession())
            {
                #region put_sample
                // Create a new document
                var doc = new Category
                {
                    Name = "My category",
                    Description = "My category description"
                };
                // Create metadata on the document
                var docInfo = new DocumentInfo
                {
                    Collection = "Categories"
                };
                // Convert your entity to a BlittableJsonReaderObject
                var blittableDoc = session.Advanced.JsonConverter.ToBlittable(doc, docInfo);

                // The Put command (parameters are document ID, changeVector check is null, the document to store)
                var command = new PutDocumentCommand(null, "categories/999", null, blittableDoc);

                // RequestExecutor sends the command to the server
                session.Advanced.RequestExecutor.Execute(command, session.Advanced.Context);
                #endregion
            }
        }
    }
}

File: Documentation/5.4/Samples/nodejs/client-api/commands/documents/put.js
import { DocumentStore } from "ravendb";
import assert from "assert";

const documentStore = new DocumentStore();
const session = documentStore.openSession();

async function putDocumentsCommand() {
    {
        //region put
        // Create a new entity
        const category = new Category();
        category.name = "My category";
        category.description = "My category description";
       
        // To be able to specify under which collection the document should be stored 
        // you need to convert the entity to a json document first.
        
        // Passing the entity as is instead of the json document
        // will result in storing the documet under the "@empty" collection.
        
        const documentInfo = new DocumentInfo();
        documentInfo.collection = "categories"; // The target collection
        const jsonDocument = EntityToJson.convertEntityToJson(category, documentStore.conventions, documentInfo);
        
        // Define the Put Command
        // Pass the document ID, whether to make concurrency checks, and the json document to store
        const putDocCmd = new PutDocumentCommand("categories/999", null, jsonDocument);
        
        // Send the command to the server using the RequestExecutor
        await documentStore.getRequestExecutor().execute(putDocCmd);

        // Can check the command's result
        const result = putDocCmd.result;
        assert.strictEqual(result.id, "categories/999");
        //endregion
    }
}

{
    //region syntax_1
    PutDocumentCommand(id, changeVector, document);
    //endregion
    
    //region syntax_2
    // Executing the `PutDocumentCommand` returns the following object:
    {
        // string, the document id under which the entity was stored
        id;
        
        // string, the change vector assigned to the stored document
        changeVector;
    }
    //endregion
}
